snippets.json

```
@@ -0,0 +1,356 @@
+{
+    "For Loop (Collection)": {
+        "prefix": [
+            "for-each"
+        ],
+        "body": [
+            "for(${1:x} in ${2:iter}.vals()) {",
+            "\t$0",
+            "};"
+        ],
+        "description": "A for loop which iterates through a collection."
+    },
+    "For Loop (Range)": {
+        "prefix": [
+            "for-range"
+        ],
+        "body": [
+            "for(${1:i} in Iter.range(0, ${2:count} - 1)) {",
+            "\t$0",
+            "};"
+        ],
+        "description": "A for loop over a numeric range."
+    },
+    "Simple Type": {
+        "prefix": [
+            "type"
+        ],
+        "body": [
+            "type ${1:Name} = ${2:declaration};"
+        ],
+        "description": "Simple type declaration"
+    },
+    "Object Type": {
+        "prefix": [
+            "type-obj"
+        ],
+        "body": [
+            "type ${1:Name} = {",
+            "\t${2:attribute} : ${3:Type};",
+            "};"
+        ],
+        "description": "an object type block"
+    },
+    "Type Variant": {
+        "prefix": [
+            "type-variant"
+        ],
+        "body": [
+            "type ${1:Name} = {",
+            "\t#${2:Attribute};",
+            "};"
+        ],
+        "description": "a variant type block"
+    },
+    "Mutable Variable": {
+        "prefix": "var",
+        "body": [
+            "var ${1} = ${2};"
+        ],
+        "description": "declares a mutable variable"
+    },
+    "Immutable Variable": {
+        "prefix": "let",
+        "body": [
+            "let ${1} = ${2};"
+        ],
+        "description": "declares an immutable variable"
+    },
+    "Simple Function": {
+        "prefix": [
+            "func"
+        ],
+        "body": [
+            "func ${1:name} (4{2:arg} : ${5:type}) : ${4:returnType} {",
+            "\t${5};",
+            "};"
+        ],
+        "description": "Simple function declaration"
+    },
+    "Actor Function": {
+        "prefix": [
+            "actor-func"
+        ],
+        "body": [
+            "public func ${1:name} (4{2:arg} : ${5:type}) : async ${4:returnType} {",
+            "\t${5};",
+            "};"
+        ],
+        "description": "Public actor function declaration"
+    },
+    "Base Library Import": {
+        "prefix": [
+            "base-import"
```

```suggestion
            "import-base"
```

Since the prompt pops up when you start typing, it's better to start with the keyword that someone will organically start typing